src/main/resources/liquibase/202002061140-update-copy-measurement-procedure.xml

```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="202002061140-1" author="mstead" dbms="postgresql">
+        <createProcedure>
+            CREATE OR REPLACE FUNCTION copy_measurement() RETURNS trigger AS $$
+            BEGIN
+                IF (NEW.measurement_type = 'TOTAL') THEN
+                    UPDATE tally_snapshots SET
+                        cores = NEW.cores,
+                        instance_count = NEW.instance_count,
+                        sockets = NEW.sockets
+                    WHERE id = NEW.snapshot_id;
+                ELSIF (NEW.measurement_type = 'PHYSICAL') THEN
+                    UPDATE tally_snapshots SET
+                        physical_cores = NEW.cores,
+                        physical_instance_count = NEW.instance_count,
+                        physical_sockets = NEW.sockets
+                    WHERE id = NEW.snapshot_id;
+                ELSIF (NEW.measurement_type = 'HYPERVISOR') THEN
+                    UPDATE tally_snapshots SET
+                        hypervisor_cores = NEW.cores,
+                        hypervisor_instance_count = NEW.instance_count,
+                        hypervisor_sockets = NEW.sockets
+                    WHERE id = NEW.snapshot_id;
+                ELSIF (not NEW.measurement_type = ANY('{AWS,GOOGLE,AZURE,ALIBABA}'::text[])) THEN
```

Comment:
Was this change necessary as the exception causes the insert to be rolled back otherwise?

Reply:
When the job runs, if it comes across a host with a cloud provider, it blows up when this procedure runs and causes the entire job to stop.